Dockerfile

```diff
@@ -2,33 +2,34 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-FROM alpine:3.6
+FROM python:3.8.3-slim-buster
 
-RUN addgroup -g 10001 app; \
-    adduser -D -u 10001 -G app -g app app; \
-    mkdir /app; \
-    chown app:app /app
+ENV USER=app
+ENV UID=10001
 
-ENV HG_VERSION=4.4
+RUN adduser --disabled-password --uid "$UID" --gecos "" "$USER"; \
+    mkdir /"$USER"; \
+    chown "$USER":"$USER" /"$USER"
 
-RUN apk update; \
-    apk add --no-cache python2 ca-certificates; \
-    apk add --no-cache --virtual build-dependencies build-base python-dev py-pip; \
-    pip install --no-cache "mercurial>=$HG_VERSION,<$HG_VERSION.99"; \
+ENV HG_VERSION=5.3.2
+
+RUN apt-get update; \
+    apt-get install -y build-essential; \
+    pip install --no-cache "mercurial==$HG_VERSION"; \
     mkdir -p /etc/mercurial
 
 COPY hgrc /etc/mercurial/hgrc
 COPY requirements.txt /requirements.txt
-COPY entrypoint.sh /entrypoint.sh
-COPY version.json /app/version.json
+COPY entrypoint /entrypoint
+COPY version.json /"$USER"/version.json
```

Comment:
Nit: do we need quotes around `$USER` everywhere? I _think_ you can just use `$USER` in most places.

Reply:
I think you are right, let me update.